File: interface.py
from pieces import PieceType

def pos_to_square(pos):
    return chr(97 + pos[1]) + str(7 - pos[0] + 1)

def square_to_pos(square):
    return [7 - int(square[1]) + 1, ord(square[0]) - 97]

def move_to_uci(origin_pos, target_pos, board):
    if board[origin_pos[0]][origin_pos[1]].get_type() == PieceType.PAWN:
        if target_pos[0] == 0 or target_pos[0] == 7:
            return pos_to_square(origin_pos) + pos_to_square(target_pos) + "q"

    return pos_to_square(origin_pos) + pos_to_square(target_pos)

def uci_to_move(uci):
    return [square_to_pos(uci[0:2]), square_to_pos(uci[2:4])]

File: Interface.java
import pieces.Piece;
import pieces.enums.*;

public abstract class Interface {

    private static String getChar(Piece piece) {
        PieceType type = piece.getType();
        switch (type) {
            case EMPTY:
                return " ";
            case PAWN:
                return "P";
            case KNIGHT:
                return "N";
            case BISHOP:
                return "B";
            case ROOK:
                return "R";
            case QUEEN:
                return "Q";
            case KING:
                return "K";
        }
        
        throw new UnsupportedOperationException("!! UNSUPPORTED PIECETYPE !!");
    }

    public static void printBoard(Piece[][] board) {
        System.out.println("    0   1   2   3   4   5   6   7");
        System.out.println("  +---+---+---+---+---+---+---+---+");

        for (int i = 0; i < 8; i++) {
            String line = i + " | ";

            for (Piece col : board[i]) {
                
                String piece = " ";
                PieceColor color = col.getColor();

                piece = getChar(col);

                if (color == PieceColor.BLACK) {
                    piece = piece.toLowerCase();
                }

                line += piece + " | ";

            }

            System.out.println(line);
            System.out.println("  +---+---+---+---+---+---+---+---+");
        }
    }

    public static String posToSquare(int[] pos) {
        return (char) (97 + pos[1]) + Integer.toString(8 - pos[0]);
    }

    public static int[] squareToPos(String square) {
        return new int[] {8 - (square.charAt(1) - '0'), square.charAt(0) - 'a'};
    }

    public static String moveToUci(int[] origin_pos, int[] target_pos) {
        return posToSquare(origin_pos) + posToSquare(target_pos);
    }

    public static int[][] uciToMove(String uci) {
        return new int[][] {squareToPos(uci.substring(0, 2)), squareToPos(uci.substring(2, 4))};
    }
}